File: core/src/main/java/com/uiptv/model/AccountStatus.java
package com.uiptv.model;

import java.util.Locale;

public enum AccountStatus {
    ACTIVE,
    SUSPENDED;

    public static AccountStatus fromValue(String value) {
        if (value == null || value.isBlank()) {
            return null;
        }
        String normalized = value.trim().toUpperCase(Locale.ROOT);
        try {
            return AccountStatus.valueOf(normalized);
        } catch (IllegalArgumentException _) {
            return null;
        }
    }

    public String toDisplay() {
        return name().toLowerCase(Locale.ROOT);
    }
}

File: server/src/main/kotlin/com/uiptv/model/AccountStatus.kt
package com.uiptv.model

import java.util.Locale

enum class AccountStatus {
    ACTIVE,
    SUSPENDED;

    fun toDisplay(): String = name.lowercase(Locale.ROOT)

    companion object {
        @JvmStatic
        fun fromValue(value: String?): AccountStatus? {
            if (value.isNullOrBlank()) {
                return null
            }
            return try {
                valueOf(value.trim().uppercase(Locale.ROOT))
            } catch (_: IllegalArgumentException) {
                null
            }
        }
    }
}
